src/protego.py

```diff
@@ -97,20 +108,59 @@ def applies_to(self, robotname):
             return len(self.user_agent)
         return 0
 
-    def _quote_path(self, path, safe='/'):
+    def _unquote(self, url, ignore='', errors='replace'):
+        """Replace %xy escapes by their single-character equivalent."""
+        if '%' not in url:
+            return url
+
+        def hex_to_byte(h):
+            """Replaces a %xx escape with equivalent binary sequence."""
+            if six.PY3:
```

Comment:
Have the if for the Python 2 code instead. This code, as it is, will not work in Python 4 :)